File: fizzbuzz/fizzbuzz.js
const range = (i, j) => [...Array(j + 1).keys()].filter(x => x >= i);

const fizzbuzz = (i) => {
  const filter = (j) => {
    const dividesp = y => (j % y) === 0;
    if (dividesp(15)) { return 'FizzBuzz'; }
    if (dividesp(5)) { return 'Buzz'; }
    if (dividesp(3)) { return 'Fizz'; }
    return j;
  };

  range(1, i).forEach(x => console.log(filter(x)));
};

fizzbuzz(100);

File: fizzbuzz/fizzbuzz.ml
let range i j = List.init j (fun x -> x + i);;
let rec gcd i j = if j = 0 then abs i else gcd j (i mod j);;
let (>>) f g x = g(f(x));;

let fizzbuzz i =
  let filter j =
    match (gcd j 15) with
    | 15 -> "FizzBuzz"
    | 5  -> "Buzz"
    | 3  -> "Fizz"
    | _  -> string_of_int j
  in range 1 i |> List.iter (filter >> (Printf.printf "%s\n"));;

fizzbuzz 100

